frontend/packages/schema-editor/src/components/SchemaInspector/ItemDataComponent.tsx

```diff
@@ -63,15 +63,18 @@ export function ItemDataComponent({ schemaNode }: IItemDataComponentProps) {
   } = schemaNode;
   const dispatch = useDispatch();
   const { data, save, setSelectedTypePointer } = useSchemaEditorAppContext();
+  const { t } = useTranslation();
 
   const [itemTitle, setItemItemTitle] = useState<string>(title || '');
+  const [itemDescription, setItemItemDescription] = useState<string>(description || '');
   const [nodeName, setNodeName] = useState(getNameFromPointer({ pointer }));
 
   useEffect(() => {
     setNodeName(getNameFromPointer({ pointer }));
-  }, [pointer]);
+    setItemItemTitle(title);
+    setItemItemDescription(description);
```

Comment:
I have tested and this fixes the issue, but I can't see why it should be necessary to do this. It should be enough to assign the initial value to the use state itself. 🤔 React should re-render this component if some of the props change. Here the title will be changed from between value and undefined.

I will be more than happy to try to debug this tomorrow. 😊

Reply:
Yep, I totally agree 😅 It was an extremely frustrating issue to work on since it just did not make sense... As i debugged it I was convinced that it should have worked as it was...